src/serialization/interval_log/mod.rs

```diff
@@ -12,58 +12,102 @@
+//! opened. It still uses a StartTime of 1500000040 because we want to represent that this log
+//! pertains to something that started an hour ago, but we'll use a BaseTime of 1500003640 so that
+//! our delta timestamps start over at 0. Again, you could just as well decide to use the same
+//! BaseTime of 1500000040 if you prefer, but it's arguably easier for a human to parse "h hours
+//! into the benchmark, s seconds past the hour" than it is to mentally divide by 3600, so we'll
+//! go with the new BaseTime.
+//!
+//! Suppose now you wanted to process these logs and show information (e.g. the 0.999th quantile of
+//! each interval) as the y-axis on a plot where time is the x-axis. You would want to have
+//! StartTime be the zero of the x-axis. For each interval, calculate the Unix time by adding its
+//! timestamp to BaseTime, where BaseTime is assumed to be zero if it's not specified. The point on
+//! the x-axis for that interval would be the result of subtracting StartTime from that Unix time.
+//! As an example, the 17th minute in the 2nd hour would have an interval timestamp of approximately
+//! 1020, which when added to 1500003640 is 1500004660. The difference between StartTime and then is
+//! 4620. You might choose to display that in units of minutes, which would be 4620 / 60 = 77.
+//!
+//! #### Java interop
+//!
+//! Since you may be interoperating with the Java reference implementation, here's a summary of the
+//! logic used by the `HistogramLogReader` class for StartTime and BaseTime. It's slightly different
+//! than what was described above, presumably for legacy compatibility reasons. This class stores
+//! the StartTime as a field which is exposed via a getter, and also integrates filtering intervals
+//! based on windows for either "absolute" (Unix time) timestamps or "offset" timestamps (delta vs
+//! the StartTime), so if you're using that filtering functionality, you need to understand how it
+//! ends up setting its internal version of StartTime.
+//!
+//! - Neither StartTime nor BaseTime are present: interval timestamps are interpreted as seconds
+//! since the epoch. The first interval's timestamp is stored to the StartTime field.
 //! - StartTime is present: StartTime is a number of seconds since epoch, and interval timestamps
-//! should be interpreted as deltas that could be added to StartTime if seconds since epoch for each
-//! interval is needed.
-//! - BaseTime is present: same as the case where StartTime is present. It's seconds since epoch,
-//! with interval timestamps as deltas.
+//! may be interpreted as deltas to be added to StartTime or as "absolute" Unix time depending on a
+//! heuristic. In other words, the heuristic chooses between setting the effective BaseTime to 0 or
+//! to StartTime.
+//! - BaseTime is present: BaseTime is a number of seconds since epoch, and interval timestamps are
+//! interpreted as deltas. The first interval's timestamp is stored to the StartTime field.
```

Comment:
Hmm.. So `StartTime` is also relative to `BaseTime`? Sort of?

Reply:
This kind of thing is why I didn't want to make a higher level iterator that's an equivalent to HistogramLogReader until we get some clarity from users about what parts of its behavior they actually use. :)